cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -417,9 +426,8 @@ where
 		// Selection policy:
 		// - minimize the transaction equivocation.
 		// - maximize the executor computation power.
-		let mut extrinsics = {
-			let inherent_data = self.inherent_data(parent_hash, primary_hash).await?;
-
+		let mut extrinsics = if self.to_include_inherents.load(Ordering::Relaxed) {
```

Comment:
There is an atomic swap method so you don't have to have load and store separately